components/pagerduty/pagerduty.app.mjs

```diff
@@ -0,0 +1,477 @@
+      optional: true,
+      async options({ prevContext }) {
+        const {
+          offset: prevOffset,
+          more: prevMore,
+        } = prevContext;
+
+        if (prevMore === false) {
+          return [];
+        }
+
+        const {
+          escalation_policies: escalationPolicies,
+          offset,
+          more,
+        } =
+          await this.listEscalationPolicies({
+            params: {
+              offset: prevOffset,
+            },
+          });
+
+        const options = escalationPolicies.map((policy) => {
+          return {
+            label: policy.summary,
+            value: policy.id,
+          };
+        });
+
+        return {
+          options,
+          context: {
+            offset,
+            more,
+          },
+        };
+      },
+    },
+    serviceId: {
+      type: "string",
+      label: "Service ID",
+      description: "The ID of the service. [See the docs here](https://developer.pagerduty.com/api-reference/b3A6Mjc0ODE5Ng-list-services)",
+      async options({ prevContext }) {
+        const {
+          offset: prevOffset,
+          more: prevMore,
+        } = prevContext;
+
+        if (prevMore === false) {
+          return [];
+        }
+
+        const {
+          services,
+          offset,
+          more,
+        } =
+          await this.listServices({
+            params: {
+              offset: prevOffset,
+            },
+          });
+
+        const options = services.map((service) => ({
+          label: service.name,
+          value: service.id,
+        }));
+
+        return {
+          options,
+          context: {
+            offset,
+            more,
+          },
+        };
+      },
+    },
+    incidentId: {
+      type: "string",
+      label: "Incident ID",
+      description: "The ID of the incident. [See the docs here](https://developer.pagerduty.com/api-reference/b3A6Mjc0ODEzOA-list-incidents)",
+      async options({
+        statuses,
+        prevContext,
+      }) {
+        const {
+          offset: prevOffset,
+          more: prevMore,
+        } = prevContext;
+
+        if (prevMore === false) {
+          return [];
+        }
+
+        const {
+          incidents,
+          offset,
+          more,
+        } =
+          await this.listIncidents({
+            params: {
+              offset: prevOffset,
+              statuses,
+            },
+          });
+
+        const options = incidents.map((incident) => ({
+          label: incident.summary,
+          value: incident.id,
+        }));
+
+        return {
+          options,
+          context: {
+            offset,
+            more,
+            statuses,
+          },
+        };
+      },
+    },
+    userId: {
+      type: "string",
+      label: "User ID",
+      description: "The ID of the User. [See the docs here](https://developer.pagerduty.com/api-reference/b3A6Mjc0ODIzMw-list-users)",
+      optional: true,
```

Comment:
See comment on `userId` in [find-oncall-user.mjs](https://github.com/PipedreamHQ/pipedream/pull/1982/files#r831196948)

> Since this prop is in general probably required, would it be better to remove the optional: true from the app and also remove this line? See the following comment: https://github.com/PipedreamHQ/pipedream/pull/2013#discussion_r779210415

```suggestion
```